### PR TITLE
allow matching and filtering traces

### PR DESCRIPTION
**Pull request by OneOfOne**
This PR adds 2 new flags:
```
-f `regexp`, hides trace headers that matches the regexp.
-m `regexp`, shows only the trace headers that matches the regexp.
```

Example:

```
➤ go run main.go -m 'PutTask|RemoveTask' ../../xxx/xxx/crash.log
1: semacquire [26 minutes] [Created by reportQueue.PutTask @ q.go:104]
    sync          sema.go:297                 runtime_notifyListWait(#343, #735)
    sync          cond.go:57                  (*Cond).Wait(#342)
    io            pipe.go:47                  (*pipe).read(#341, #348, 0x8000, 0x8000, 0, 0, 0)
    io            pipe.go:130                 (*PipeReader).Read(#588, #348, 0x8000, 0x8000, 0x8000, 0x8000, #13)
    io            io.go:390                   copyBuffer(#1482, #1125, #7, #588, #348, 0x8000, 0x8000, #1482, 0x4a80ac, 0x25)
    io            io.go:360                   Copy(#1482, #1125, #7, #588, 0xbdc020, 0, #1482)
    iodb          bucket.go:206               (*bucket).PutTimed.func1(#1482, #1125, #1125, #1482)
    iodb          bucket.go:167               (*bucket).PutTimedFunc(#62, #298, 0x20, #1124, #1383, 0, 0, 0, 0, 0)
    iodb          bucket.go:207               (*bucket).PutTimed(#62, #298, 0x20, #7, #588, #1383, 0, 0, 0, 0xa15711, ...)
    reportQueue   q.go:99                     PutTask.func1(#437, #1383, #395, 0x4, #1193, #303, #44)

```

### CODE REVIEW

File: internal/main.go [hunk 1]
```diff
@@ -100,13 +111,38 @@ func Main() error {
 	forceColor := flag.Bool("force-color", false, "Forcibly enable coloring when with stdout is redirected")
 	parse := flag.Bool("parse", true, "Parses source files to deduct types; use -parse=false to work around bugs in source parser")
 	verboseFlag := flag.Bool("v", false, "Enables verbose logging output")
+	filterFlag := flag.String("f", "", "A regexp to only show traces that does NOT match its, ex: -f 'IO wait|syscall'")
```
Comment (maruel) [inline internal/main.go]: "A " is superfluous

Comment (maruel) [inline internal/main.go]: Also it's not clear to the reader of the help page what the regexp matches, the raw string, the package?

File: internal/main.go [hunk 2]
```diff
@@ -27,6 +27,8 @@ import (
 	"os/signal"
 	"syscall"
 
+	"regexp"
```
Comment (maruel) [inline internal/main.go]: Why it is spaced? Please use goimports to sort.

**Comment by maruel** [comment]
Sorry for the long delay, I was at GopherCon. I took the time to fix the tests and make the one in source_test.go much more maintainble, please rebase on 
868abbf1ebac0fb2760cd9a410a5bd2f5afb2f76 so that you can add test and confirm it is green.

**Comment by maruel** [comment]
Hi! Can you address the comments above? In particular adding a unit test for the functionality so the code coverage doesn't lower. https://blog.golang.org/cover is a good reference to learn how to learn about Go coverage, which helps writing unit test to cover as many lines as possible. Thanks!

**Comment by maruel** [comment]
Ping.

**Comment by OneOfOne** [comment]
I have been crazy busy and completely forgot, but I have to say it saved me few times over the last few weeks.

I added the tests but it still says it went down, so I'm not sure what I'm missing.

**Comment by maruel** [comment]
Can you reply to my review comments? I think this feature is useful but I'd like to see these small things fixed first.

Thanks!

**Comment by OneOfOne** [comment]
Ok I must be going blind, but what comments?

I can't figure out how to keep coveralls happy, if you'd like you can merge it then change that or even fork my version and change it directly then merge that, I really don't mind.

File: internal/main.go (hunk 2 above)
Comment (maruel) [reply internal/main.go]: Please fix the import.

File: internal/main.go (hunk 1 above)
Comment (maruel) [reply internal/main.go]: Please address.

**Comment by maruel** [comment]
Ping again, as I see value in the CI. I'm open to chat over IM to figure the issues out.

**Comment by OneOfOne** [comment]
I'll try to get it done today, sorry I hadn't noticed your last ping.

**Comment by maruel** [comment]
You can join on https://gitter.im/maruel/panicparse

**Comment by maruel** [comment]
Thanks for the contribution

I committed this change squashed but kept your authorship, and added fixes for change since the PR was created. I'm still ambivalent about the flags and it needs to be supported for HTML output, so there's a fair chunk of follow up work left but I'll do this in separate changes.

**Pull request closed without merging**